olive/auto_optimizer/__init__.py

```diff
@@ -0,0 +1,126 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
+
+import logging
+from copy import deepcopy
+from enum import Enum
+from typing import Dict, List, Optional
+
+from olive.auto_optimizer.regulate_mixins import RegulatePassConfigMixin
+from olive.common.config_utils import ConfigBase
+from olive.common.pydantic_v1 import validator
+from olive.data.config import DataConfig
+from olive.evaluator.olive_evaluator import OliveEvaluatorConfig
+from olive.hardware.accelerator import AcceleratorSpec
+from olive.model import ModelConfig
+
+logger = logging.getLogger(__name__)
+
+
+class Precision(str, Enum):
+    FP32 = "fp32"
+    FP16 = "fp16"
+    INT8 = "int8"
+    INT4 = "int4"
+
+
+class AutoOptimizerConfig(ConfigBase):
+    # opt_level
+    # 0: look up template mapping
+    # TODO(anyone): support more opt_level
+    opt_level: int = 0
```

Comment:
For other optimization tools, `opt_level` usually means different precision. Do we want to follow the same concept here? Or will define a new `opt_level` for Olive.

Reply:
We don't need to imply that opt_level impacts precision.

Reply:
@devang-ml what does `opt_level` in Olive mean? Does it mean different pass flow?

Reply:
yes

Reply:
Updated the comments. Or do you strongly suggest to update the variable name (opt_level -> ? )?